gateleen-hook/src/main/java/org/swisspush/gateleen/hook/HookHandler.java

```diff
@@ -1032,23 +1041,33 @@ private void handleListenerRegistration(final HttpServerRequest request) {
             final String listenerStorageUri = hookRootUri + HOOK_LISTENER_STORAGE_PATH + getUniqueListenerId(request.uri());
 
             // Extract expireAfter from the registration header.
-            final String myExpireAfter = request.headers().get( X_EXPIRE_AFTER );
             final String expirationTime;
-            if( "-1".equals(myExpireAfter) ){
-                expirationTime = EXPIRATION_TIME_NEVER;
-            }else{
-                Integer expireAfter = ExpiryCheckHandler.getExpireAfter(request.headers());
-                if (expireAfter == null) {
+
+            { // Evaluate expiration time
+                Integer expireAfter;
+                try{
+                    expireAfter = ExpiryCheckHandler.getExpireAfter2( request.headers() );
```

Comment:
The _getExpireAfter2()_ method should not throw exceptions (like the other methods in this class do neither). You could for example return a _java.util.Optional_ object which would be _Optional.absent()_ in error case

Reply:
How then a caller (eg HookHandler:1049) should distinguish between _no-header-using-default_ (Line 1051) and _infinite_ (Line 1060) if both get returned by null?

Reply:
Because absent (aka null) is already used to indicate _header-not-present_
Cited from javadoc of ExpiryCheckHandler#getExpireValue(String):
> _returns Expire-after time in seconds_ **or null if header not present**.

Actually non existent header is not an error (or did I miss a specification somewhere?). Therefore a caller can decide to use a default value in this case (See: HookHandler:1051).

Where in contrast a X-Expire-After value eg "-1234" or "Hello World" would definitively be a client Problem and therefore should be answered with a 400 response code (In my opinion at least. No spec found which declares type of X-Expire-After).

If we Cut down this API using null (or absent() (doesn't matter)) for both (not-present AND error), then all callers (especially future ones) will never be able to differ those cases. Therefore we'll be forced to extend our API again, again and again violating Interface-Segregation-Principle [1] (or also MinimalInterface [2]) more and more.

Another (more verbose) variant would be to throw a checked exception like MyCheckedNumberFormatException so we'll force callers to handle that case explicitly.

Never mind. I'll try to replace it with a more limited interface  :-)

[1] https://en.wikipedia.org/wiki/Interface_segregation_principle
[2] https://martinfowler.com/bliki/MinimalInterface.html